warehouse/query-core/src/test/java/datawave/query/jexl/visitors/IvaratorRequiredVisitorTest.java

```diff
@@ -0,0 +1,34 @@
+package datawave.query.jexl.visitors;
+
+import datawave.query.jexl.JexlASTHelper;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.ParseException;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class IvaratorRequiredVisitorTest {
```

Comment:
This is a good start. At a minimum I would like to see some additional test cases for when the ivarator node is the child of an intersection and the child of a union
```
# base case
((_Value_ = true) && (FOO =~ 'ba.*'))

# with intersection
FOO == 'bar' && ((_Value_ = true) && (FOO =~ 'ba.*'))

# with union
FOO == 'bar' || ((_Value_ = true) && (FOO =~ 'ba.*'))
```

At that point it makes sense to refactor your test to have a common test method that accepts a query and the expected result of a visit. 
```
private void test(String query, boolean expected){
// common code here
```

Reply:
Good idea - will make those changes